packages/integrations/src/vue.ts

```diff
@@ -1,103 +1,298 @@
 import { EventProcessor, Hub, Integration } from '@sentry/types';
-import { getGlobalObject, isPlainObject, logger } from '@sentry/utils';
+import { basename, getGlobalObject, logger, timestampWithMs } from '@sentry/utils';
+import { Integrations as APMIntegrations, Span as SpanClass } from '@sentry/apm';
+
+interface IntegrationOptions {
+  Vue: any;
+  /**
+   * When set to false, Sentry will suppress reporting of all props data
+   * from your Vue components for privacy concerns.
+   */
+  attachProps: boolean;
+  /**
+   * When set to true, original Vue's `logError` will be called as well.
+   * https://github.com/vuejs/vue/blob/c2b1cfe9ccd08835f2d99f6ce60f67b4de55187f/src/core/util/error.js#L38-L48
+   */
+  logErrors: boolean;
+  tracing: boolean;
+  tracingOptions: TracingOptions;
+}
+
+interface TracingOptions {
+  track: boolean | Array<string>;
```

Comment:
Can we make the naming more verbose like: `trackComponents`, also let's add descriptions so user get an explanation in IntelliSense.

Reply:
Done